From 83f7ee3c6080aa70265f4cdd6855f0b13d48d76d Mon Sep 17 00:00:00 2001
From: Frank Wunderlich <frank-w@public-files.de>
Date: Mon, 2 Dec 2024 16:42:54 +0100
Subject: [PATCH 70/93] ci: try to fix build-error caused by change for
 multiarch libssl

e2c318225ac1 kbuild: deb-pkg: add pkg.linux-upstream.nokernelheaders build profile
---
 .github/workflows/build.yml | 43 ++++++++++++++++++++++++++++---------
 1 file changed, 33 insertions(+), 10 deletions(-)

diff --git a/.github/workflows/build.yml b/.github/workflows/build.yml
index 763899315aec..6d65c698e1cc 100644
--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -25,7 +25,8 @@ jobs:
     permissions:
       contents: write  # for softprops/action-gh-release to create GitHub release
     # The type of runner that the job will run on
-    runs-on: ubuntu-latest
+    #runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
 
     # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
@@ -35,16 +36,42 @@ jobs:
         with:
           fetch-depth: 10
 
-      - name: Install depencies
-        run: |
-          sudo apt update
-          sudo apt install ccache libssl-dev u-boot-tools python3-mako debhelper fakeroot gcc-arm-linux-gnueabihf gcc-aarch64-linux-gnu make device-tree-compiler libncurses5-dev libelf-dev
-
       - name: Setup env
         run: |
           echo "DT=$(date +'%Y-%m-%d_%H%M')" >> $GITHUB_ENV
           echo "KERNELVER=$(make kernelversion)" >> $GITHUB_ENV
           echo "BRANCH=${GITHUB_REF##*/}" >> $GITHUB_ENV
+          echo "UBUNTU_MAJOR_VERSION=$(cat /etc/issue | head -1|sed -e 's/^Ubuntu \([0-9]\+\).*$/\1/')" >> $GITHUB_ENV
+
+      - name: Print env
+        run: |
+          echo $BRANCH $KERNELVER $DT
+          echo $UBUNTU_MAJOR_VERSION
+
+      - name: update apt-repos (u22)
+        if: env.UBUNTU_MAJOR_VERSION == '22'
+        run: |
+          cat /etc/apt/sources.list
+          sudo sed -i.bak -e 's/^deb/deb [ arch=amd64,i386 ]/' /etc/apt/sources.list
+          #deb [arch=amd64,i386] http://archive.ubuntu.com/ubuntu/ jammy main universe
+          echo "deb [arch=arm64,armhf] http://ports.ubuntu.com/ubuntu-ports jammy main universe" | sudo tee -a /etc/apt/sources.list
+
+
+      - name: update apt-repos (u24)
+        if: env.UBUNTU_MAJOR_VERSION == '24'
+        run: |
+          sudo cp /etc/apt/sources.list.d/ubuntu.sources /etc/apt/sources.list.d/ports.sources
+          sudo sed -i.bak -e 's/^\(Suites:.*\)$/\1\nArchitectures: amd64 i386/' /etc/apt/sources.list.d/ubuntu.sources
+          sudo sed -i.bak -e 's/^URIs:.*$/URIs: http:\/\/ports.ubuntu.com\/ubuntu-ports\//' /etc/apt/sources.list.d/ports.sources
+          sudo sed -i -e 's/^\(Suites:.*\)$/\1\nArchitectures: armhf arm64/' /etc/apt/sources.list.d/ports.sources
+          cat /etc/apt/sources.list.d/ports.sources
+
+      - name: Install depencies
+        run: |
+          sudo dpkg --add-architecture armhf
+          sudo dpkg --add-architecture arm64
+          sudo apt update
+          sudo apt install ccache libssl-dev:armhf libssl-dev:arm64 build-essential u-boot-tools python3-mako debhelper fakeroot gcc-arm-linux-gnueabihf gcc-aarch64-linux-gnu make device-tree-compiler libncurses5-dev libelf-dev
 
       - name: Generate Changelog
         run: |
@@ -52,10 +79,6 @@ jobs:
           echo "last commits:" >> ${{ github.workspace }}-CHANGELOG.txt
           git log --pretty=format:"%h %ad %s %d by %an" --date=short >> ${{ github.workspace }}-CHANGELOG.txt
 
-      - name: Print env
-        run: |
-          echo $BRANCH $KERNELVER $DT
-
       - name: Setup cache
         id: cache
         uses: actions/cache@v4
-- 
2.49.0

